jetty-client/src/main/java/org/eclipse/jetty/client/AuthenticationProtocolHandler.java

```diff
@@ -82,80 +77,71 @@ protected HttpClient getHttpClient()
         // Return new instances every time to keep track of the response content
         return new AuthenticationListener();
     }
-    
 
-    
-    protected List<HeaderInfo> getHeaderInfo(String value) throws IllegalArgumentException
+
+    protected List<HeaderInfo> getHeaderInfo(String header) throws IllegalArgumentException
     {
-        String header = value;
         List<HeaderInfo> headerInfos = new ArrayList<>();
-        
-        while(true)
+
+        Matcher m;
+        String authScheme = null;
+        Map<String,String> authParams = new HashMap<>();
+
+        for(String value : new QuotedCSV(true, header))
         {
-            Matcher m = MULTIPLE_CHALLENGE_PATTERN.matcher(header);
+            m = CHALLENGE_PATTERN.matcher(value);
             if (m.matches())
             {
-                headerInfos.add(newHeaderInfo(m.group(1)));
-                header = m.group(2);
-            }
-            else
-            {
-                headerInfos.add(newHeaderInfo(header));
-                break;
+                if(m.group("schemeOnly") != null)
+                {
+                    if (authScheme != null)
+                    {
+                        headerInfos.add(new HeaderInfo(getAuthorizationHeader(), authScheme, authParams));
```

Comment:
Ah, this is adding the last scheme as a `HeaderInfo` before remembering the next. That is not immediately obvious from first glance, so a comment to that effect would be good.

Alternately, if the param map within `HeaderInfo` is mutable, just create and add the current scheme in a `HeaderInfo` here and then add params to the map as they are discovered.